cmd/erasure-object.go

```diff
@@ -159,6 +160,15 @@ func (er erasureObjects) GetObjectNInfo(ctx context.Context, bucket, object stri
 
 	fi, metaArr, onlineDisks, err := er.getObjectFileInfo(ctx, bucket, object, opts)
 	if err != nil {
+		if errors.Is(err, errFileNotFound) ||
+			errors.Is(err, errFileVersionNotFound) {
+			// proxy to replication target if active-active replication is in place.
+			reader, proxy := proxyGetToReplicationTarget(ctx, bucket, object, rs, h, opts)
+			if !proxy || reader == nil {
+				return nil, toObjectErr(err, bucket, object)
+			}
+			return reader, nil
+		}
```

Comment:
We cannot allow this level of fallback in the hot path such as GetObject() we should not even enter the function without checking a server config setting.

Reply:
this proxying will be attempted only if the object is missing on disk in the current cluster. the proxyGetToReplicationTarget returns right away if replication config is missing. Kept all proxying related code within this function rather than bloat erasure-object.go

Reply:
Current but this codepath although gets executed during 404's - my main issue with this is that the function is loaded and it can be a non-op based on some other external data structure.  This is confusing to read.

The better way is to have an `if proxiable { proxy() }`  kind of coding style where `proxiable` is definitive - Go wouldn't then pass around each function with some values which never get used.

Reply:
done